cli/popper/commands/cmd_add.py

```diff
@@ -47,16 +45,35 @@ def cli(ctx, pipeline, folder, branch):
 
     owner, repo, pipe_name = pipeline.split('/')
 
+    if folder:
+        path, basename = os.path.split(folder)
+        if not basename:  # Only true when trailing slash is present
+            path, basename = os.path.split(path)
+        if basename:
+            new_pipe_name = basename  # Use basename as the name of pipeline
+        else:
+            pu.fail("The path is invalid.")
+    else:  # If no pipeline name is provided, use the original pipeline name
+        new_pipe_name = pipe_name
+
     config = pu.read_config()
 
-    if pipe_name in config['pipelines']:
-        pu.fail("Pipeline {} already in repo.".format(pipe_name))
+    if new_pipe_name in config['pipelines']:
+        pu.fail("Pipeline {} already in repo.".format(new_pipe_name))
 
     project_root = pu.get_project_root()
+
+    if not folder:  # Put it in the pipelines directory, default case
+        folder = os.path.join('pipelines', new_pipe_name)
+    elif '/' not in folder:  # Put it in the path inside pipelines directory
+        folder = os.path.join('pipelines', folder)
```

Comment:
with a `utils.get_name_and_path_for_new_pipeline()` function the above won't be needed